netplan_cli/cli/sriov.py

```diff
@@ -212,37 +211,47 @@ def unbind_vfs(vfs: typing.Iterable[PCIDevice], driver) -> typing.Iterable[PCIDe
     return unbound_vfs
 
 
-def _get_target_interface(interfaces, np_state, pf_link, pfs):
-    if pf_link not in pfs:
-        # handle the match: syntax, get the actual device name
-        pf_dev = np_state[pf_link]
-        if pf_dev._has_match:
-            # now here it's a bit tricky
-            set_name = pf_dev.set_name
-            if set_name and set_name in interfaces:
-                # if we had a match: stanza and set-name: this means we should
-                # assume that, if found, the interface has already been
-                # renamed - use the new name
-                pfs[pf_link] = set_name
-            else:
-                for interface in interfaces:
-                    if not pf_dev._match_interface(
-                            iface_name=interface,
-                            iface_driver=utils.get_interface_driver_name(interface),
-                            iface_mac=utils.get_interface_macaddress(interface)):
-                        continue
-                    # we have a matching PF
-                    # store the matching interface in the dictionary of
-                    # active PFs, but error out if we matched more than one
-                    if pf_link in pfs:
-                        raise ConfigurationError('matched more than one interface for a PF device: %s' % pf_link)
-                    pfs[pf_link] = interface
-        else:
-            # no match field, assume entry name is the interface name
-            if pf_link in interfaces:
-                pfs[pf_link] = pf_link
+def _interface_matches(netdef: netplan.NetDefinition, interface: str) -> bool:
+    return netdef._match_interface(
+        iface_name=interface,
+        iface_driver=utils.get_interface_driver_name(interface),
+        iface_mac=utils.get_interface_macaddress(interface))
 
-    return pfs.get(pf_link, None)
+
+def _get_interface_name_for_netdef(netdef: netplan.NetDefinition) -> Optional[str]:
+    """
+    Try to match a netdef with the real system network interface.
+    Throws ConfigurationError if there is more than one match.
+    """
+    interfaces: List[str] = netifaces.interfaces()
+    if netdef._has_match:
+        # now here it's a bit tricky
+        set_name: str = netdef.set_name
+        if set_name and set_name in interfaces:
+            # if we had a match: stanza and set-name: this means we should
+            # assume that, if found, the interface has already been
+            # renamed - use the new name
+            return set_name
+        else:
+            matches: Set[str] = set()
+            # we walk through all the system interfaces to determine if there is
+            # more than one matched interface
+            for interface in interfaces:
+                if not _interface_matches(netdef, interface):
```

Comment:
question (non-blocking): Why not just call `netdef._match_interface()` directly here?

Reply:
I don't have a strong opinion here, but I don't know it looks cleaner with `netdef._match_interface` wrapped in this method...